src/Xamarin.MSBuild.Sdk/Tasks/PrepareConsolidationProject.cs

```diff
@@ -150,10 +182,19 @@ public override bool Execute ()
             CompileItems = compileItems.ToArray ();
             ProjectReferenceItems = projectReferenceItems.ToArray ();
             ReferenceItems = referenceItems.ToArray ();
+            PackageReferenceItems = packageReferenceItems.ToArray ();
             EmbeddedResourceItems = embeddedResourceItems.ToArray ();
 
             return true;
 
+            Dictionary<string, string> GetItemMetadata (ProjectItem item)
+            {
+                var itemMetadata = new Dictionary<string, string> (StringComparer.OrdinalIgnoreCase);
```

Comment:
Nit (against myself apparently, since you just moved this chunk of code): `new Dictionary<string, string>(item.Metadata.Count, StringComparer.OrdinalIgnoreCase)` ... assuming `item.Metadata` is `IReadOnlyCollection` and not just `IEnumerable`. Also _consider_ `Enumerable.ToDictionary` ... it _may_ pre-allocate for you. Also I'm sure none of it matters. Now I have to look.